examples/platform/esp32/mode-support/static-supported-modes-manager.cpp

```diff
@@ -0,0 +1,132 @@
+        VerifyOrReturnValue(ESP32Config::KeyAllocator::SupportedModesValue(keyBuf, sizeof(keyBuf), endpointId, index) ==
+                                CHIP_NO_ERROR,
+                            ModeOptionsProvider(nullptr, nullptr));
+        ESP32Config::Key modeKey(ESP32Config::kConfigNamespace_ChipFactory, keyBuf);
+        VerifyOrReturnValue(ESP32Config::ReadConfigValue(labelKey, supportedModeMode) == CHIP_NO_ERROR,
+                            ModeOptionsProvider(nullptr, nullptr));
+
+        memset(keyBuf, 0, sizeof(char) * ESP32Config::kMaxConfigKeyNameLength);
+        VerifyOrReturnValue(ESP32Config::KeyAllocator::SemanticTagsCount(keyBuf, sizeof(keyBuf), endpointId, index) ==
+                                CHIP_NO_ERROR,
+                            ModeOptionsProvider(nullptr, nullptr));
+        ESP32Config::Key stCountKey(ESP32Config::kConfigNamespace_ChipFactory, keyBuf);
+        VerifyOrReturnValue(ESP32Config::ReadConfigValue(stCountKey, semanticTagCount) == CHIP_NO_ERROR,
+                            ModeOptionsProvider(nullptr, nullptr));
+
+        SemanticTag * semanticTags = new SemanticTag[semanticTagCount];
+        for (auto stIndex = 0; stIndex < semanticTagCount; stIndex++)
+        {
+
+            uint32_t semanticTagValue   = 0;
+            uint32_t semanticTagMfgCode = 0;
+            SemanticTag tag;
+
+            memset(keyBuf, 0, sizeof(char) * ESP32Config::kMaxConfigKeyNameLength);
+            VerifyOrReturnValue(ESP32Config::KeyAllocator::SemanticTagValue(keyBuf, sizeof(keyBuf), endpointId, index, stIndex) ==
+                                    CHIP_NO_ERROR,
+                                ModeOptionsProvider(nullptr, nullptr));
+            ESP32Config::Key stValueKey(ESP32Config::kConfigNamespace_ChipFactory, keyBuf);
+            VerifyOrReturnValue(ESP32Config::ReadConfigValue(stValueKey, semanticTagValue) == CHIP_NO_ERROR,
+                                ModeOptionsProvider(nullptr, nullptr));
+
+            memset(keyBuf, 0, sizeof(char) * ESP32Config::kMaxConfigKeyNameLength);
+            VerifyOrReturnValue(ESP32Config::KeyAllocator::SemanticTagMfgCode(keyBuf, sizeof(keyBuf), endpointId, index, stIndex) ==
+                                    CHIP_NO_ERROR,
+                                ModeOptionsProvider(nullptr, nullptr));
+            ESP32Config::Key stMfgCodeKey(ESP32Config::kConfigNamespace_ChipFactory, keyBuf);
+            VerifyOrReturnValue(ESP32Config::ReadConfigValue(stMfgCodeKey, semanticTagMfgCode) == CHIP_NO_ERROR,
+                                ModeOptionsProvider(nullptr, nullptr));
+
+            tag.value             = static_cast<uint16_t>(semanticTagValue);
+            tag.mfgCode           = static_cast<chip::VendorId>(semanticTagMfgCode);
+            semanticTags[stIndex] = static_cast<const SemanticTag>(tag);
+        }
+
+        option.label        = chip::CharSpan::fromCharString(supportedModeLabel);
```

Comment:
I don't understand this part.  This is pointing `option.label` into the `supportedModeLabel` buffer.  But then we will move on to the next mode and clobber the contents of that buffer, right?  How does this actually work?